clients/src/main/java/org/apache/kafka/common/record/AbstractRecords.java

```diff
@@ -44,6 +45,17 @@ public RecordBatch firstBatch() {
         return iterator.next();
     }
 
+    @Override
+    public Optional<RecordBatch> lastBatch() {
+        Iterator<? extends RecordBatch> iterator = batches().iterator();
+
+        RecordBatch batch = null;
+        while (iterator.hasNext())
+            batch = iterator.next();
+
+        return Optional.ofNullable(batch);
```

Comment:
For my own knowledge, why aren't we using Optional.empty() when value doesn't exist and choosing to have a nullable optional? Is that something to do with scala Option compatibility?

Reply:
Never mind. My misunderstanding.